temporalio/worker/_workflow_instance.py

```diff
@@ -414,6 +419,119 @@ def _apply_cancel_workflow(
+
+                if job.run_validator and defn.validator is not None:
+                    with self._as_read_only():
+                        self._inbound.handle_update_validator(handler_input)
+                        # Re-process arguments to avoid any problems caused by user mutation of them during validation
+                        args = self._process_handler_args(
+                            job.name,
+                            job.input,
+                            defn.name,
+                            defn.arg_types,
+                            defn.dynamic_vararg,
+                        )
+                        handler_input.args = args
+
+                past_validation = True
+                # Accept the update
+                command.update_response.accepted.SetInParent()
+                command = None  # type: ignore
+
+                # Run the handler
+                success = await self._inbound.handle_update_handler(handler_input)
+                result_payloads = self._payload_converter.to_payloads([success])
+                if len(result_payloads) != 1:
+                    raise ValueError(
+                        f"Expected 1 result payload, got {len(result_payloads)}"
+                    )
+                command = self._add_command()
+                command.update_response.protocol_instance_id = job.protocol_instance_id
+                command.update_response.completed.CopyFrom(result_payloads[0])
+            except (Exception, asyncio.CancelledError) as err:
+                logger.debug(
+                    f"Update raised failure with run ID {self._info.run_id}",
+                    exc_info=True,
+                )
+                # All asyncio cancelled errors become Temporal cancelled errors
+                if isinstance(err, asyncio.CancelledError):
+                    err = temporalio.exceptions.CancelledError(
+                        f"Cancellation raised within update {err}"
+                    )
+                # Read-only issues during validation should fail the task
+                if isinstance(err, temporalio.workflow.ReadOnlyContextError):
+                    self._current_activation_error = err
+                    return
+                # Temporal errors always fail the update. Other errors fail it during validation, but the task during
+                # handling.
+                if (
+                    isinstance(err, temporalio.exceptions.FailureError)
+                    or not past_validation
+                ):
+                    if command is None:
+                        command = self._add_command()
+                        command.update_response.protocol_instance_id = (
+                            job.protocol_instance_id
+                        )
+                    self._failure_converter.to_failure(
+                        err,
+                        self._payload_converter,
+                        command.update_response.rejected.cause,
```

Comment:
```suggestion
                        command.update_response.rejected,
```

I think this has to be right on the rejection field

Reply:
I had it that way initially and it didn't work properly.

Reply:
Hrmm, I wonder why. Which part didn't work out? `rejected` is the failure and that should have the message, stack trace, etc on it. The cause is a potential error cause of this error. May need a `command.update_response.rejected.SetInParent()` if somehow it wasn't being set.

Reply:
Yeah, I needed `SetInParent` turns out. Working now.